batch/data-plane/src/test/java/com/microsoft/azure/batch/BatchIntegrationTestBase.java

```diff
@@ -0,0 +1,517 @@
+                System.getenv("CLIENT_ID"), System.getenv("APPLICATION_SECRET"), "microsoft.onmicrosoft.com", null,
+                null);
+    }
+
+    @Before
+    public void beforeMethod() throws Exception {
+        printThreadInfo(String.format("%s: %s", "beforeTest", testName.getMethodName()));
+        createClientWithInterceptor(AuthMode.SharedKey);
+    }
+
+
+    @After
+    public void afterMethod() throws IOException {
+        interceptorManager.finalizeInterceptor();
+    }
+
+    protected RestClient buildRestClient(RestClient.Builder builder) {
+        return builder.build();
+    }
+
+    protected RestClient buildPlaybackRestClient(ServiceClientCredentials credentials, String baseUrl) throws IOException {
+        return buildRestClient(new RestClient.Builder().withBaseUrl(baseUrl)
+                .withSerializerAdapter(new AzureJacksonAdapter())
+                .withResponseBuilderFactory(new ResponseBuilder.Factory() {
+                    private final AzureResponseBuilder.Factory baseFactory = new AzureResponseBuilder.Factory();
+                    @Override
+                    public <T, E extends RestException> ResponseBuilder<T, E> newInstance(SerializerAdapter<?> serializerAdapter) {
+                        return baseFactory.<T, E>newInstance(serializerAdapter).withThrowOnGet404(true);
+                    }
+                })
+                .withCredentials(credentials)
+                .withLogLevel(LogLevel.NONE)
+                .withNetworkInterceptor(new LoggingInterceptor(LogLevel.BODY_AND_HEADERS))
+                .withNetworkInterceptor(interceptorManager.initInterceptor())
+                .withInterceptor(new ResourceManagerThrottlingInterceptor()));
+    }
+
+
+    static CloudPool createIfNotExistPaaSPool(String poolId) throws Exception {
+        // Create a pool with 3 Small VMs
+        String poolVmSize = "Small";
+        int poolVmCount = 3;
+        String poolOsFamily = "4";
+        String poolOsVersion = "*";
+
+        // 10 minutes
+        long poolSteadyTimeoutInSeconds = 10 * 60 * 1000;
+
+        // Check if pool exists
+        if (!batchClient.poolOperations().existsPool(poolId)) {
+            // Use PaaS VM with Windows
+            CloudServiceConfiguration configuration = new CloudServiceConfiguration();
+            configuration.withOsFamily(poolOsFamily).withOsVersion(poolOsVersion);
+
+            List<UserAccount> userList = new ArrayList<>();
+            userList.add(new UserAccount().withName("test-user").withPassword("kt#_gahr!@aGERDXA")
+                    .withElevationLevel(ElevationLevel.ADMIN));
+            PoolAddParameter addParameter = new PoolAddParameter().withId(poolId)
+                    .withTargetDedicatedNodes(poolVmCount).withVmSize(poolVmSize)
+                    .withCloudServiceConfiguration(configuration).withUserAccounts(userList);
+            batchClient.poolOperations().createPool(addParameter);
+        }
```

Comment:
nit: should probably log if pool already exists as its possible the existing pool is in some weird state ( I don't think we did that before so up to you)

Reply:
Added the log in case pool exists.